### DISCUSSION TITLE
Potential memory leak?

### DISCUSSION THREAD

**Discussion by maxs-rose** [help-me-mom/ng-mocks#5435] [Q&A] [answered]
After running ~2K tests the JS heap reaches 1GB of memory usage and no longer goes down.

![image](https://user-images.githubusercontent.com/20226748/231575855-d4a4ce6a-8aa3-41c0-be28-acff82927987.png)

There appears to be a lot of detached dom nodes handing around that seem to be related to `ngMocksUniverse`
![image](https://user-images.githubusercontent.com/20226748/231576820-879dd86e-7ceb-4679-84fe-c6d2f7769bb1.png)

Most of the test are setup similarly to the following:
```ts
describe("some component", () => {
     ngMocks.faster();

     const someServiceMock = MockService(SomeService);

     beforeAll(() => {
         return MockBuilder(SomeComponent).mock(SomeService, someServiceMock);
     );
     
    it("should do something", () => {
         const component = MockRender(SomeComponent).point.componentInstance;
         
         component.foo();

         expect(someServiceMock.serviceMethod).toHaveBeenCalled();
    });

    /* some more it blocks */
);
```

We have tried removing `ngMocks.faster` but that doesn't seem to have any affect.

Is this behavior due to how we are using NgMocks?

**Comment by satanTime** [comment, marked as answer]
Hi there, thank you for the question. I've converted it to a bug and try to investigate next weeks where the problem might be.
Could you verify if you remove `ng-mocks` completely from test env, then you don't see the issue?

**Reply by maxs-rose** [reply]
I am not sure how I would do that without rewriting about 2K tests 😅

**Reply by satanTime** [reply]
oki-doki - I'll try myself
